connectors/tests/test_sync.py

```diff
@@ -488,10 +488,16 @@ async def test_connector_service_poll_trace_mem(
 
 @pytest.mark.asyncio
 async def test_connector_service_poll_with_entsearch(
-    mock_responses, patch_logger, patch_ping, set_env, catch_stdout
+    mock_responses, patch_logger, patch_ping, set_env  # , catch_stdout
 ):
     with mock.patch.dict(os.environ, {"ENT_SEARCH_CONFIG_PATH": ES_CONFIG}):
-        await set_server_responses(mock_responses)
+
+        def connectors_read(url, **kw):
+            assert kw["headers"]["x-elastic-auth"] == "SomeYeahValue"
```

Comment:
Is the dictionary case-insensitive for keys? As it's looking for `x-elastic-auth` instead of `X-Elastic-Auth`

Reply:
Yeah HTTP headers are case insensitive, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

Reply:
Fun fact, they _should_ be insensitive, but are not in practice. They persist case across the network, and application code may not be designed to be insensitive, and therefore casing matters. [This bit us recently with Zendesk](https://github.com/elastic/enterprise-search-team/issues/1962#issuecomment-1176676737).

Reply:
Gotcha. But all the python clients behave like this; I assume Elasicsearch works with case insensitive? I guess it's easy to verify